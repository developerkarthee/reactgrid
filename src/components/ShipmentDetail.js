import React, { Component } from 'react';
import axios from 'axios';
import dataAdapter from '../services/DataAccessService';
class ShipmentDetail extends Component {

    state = { error: 1, data: { name: '' } };
    constructor(props) {
        super(props);
        this.updateName = this.updateName.bind(this);
        this.onSave = this.onSave.bind(this);
    }
    componentDidMount() {
        const { id } = this.props.match.params;
        //read the selected shipment info from rest end point
        dataAdapter.getShipmentDetail(id).then((res) => {
            this.setState({ error: 2 });
            this.setState({ data: res.data });
        }).catch(() => {
            this.setState({ error: 3 });
        });
    }

    updateName(event) {
        this.state.data.name = event.target.value;
        this.setState({ data: this.state.data });
    }

    onSave($event) {
        $event.preventDefault();
        const { id } = this.props.match.params;
        dataAdapter.updateShipmentName(id, this.state.data.name).then(() => {
            alert("Name has been updated successfully");
        }).catch(() => {
            alert("Failed to update the name");
        });
    }

    render() {

        if (this.state.error === 2) {
            return (
                <div className="detail-main-header">
                    <div className="shipment-detail-header">
                        <h3>
                            Name : &nbsp;{this.state.data.name}
                        </h3>
                        <form>                   
                            <div className="form-group">
                            <label >Edit the Name</label>
                            <input id="name" className="form-control" type="text" onChange={this.updateName} value={this.state.data.name}></input>

                        </div>
                            <button className="btn btn-primary" onClick={this.onSave}>Save</button>
                        </form>

                    </div>

                    <h4 className="shipment-header-text">Cargo List</h4>

                    <div className="card-group">
                        {this.state.data.cargo && this.state.data.cargo.map((item, i) => {
                            return (<div className="card" key={i}>
                                <div className="frieght-card-header card-header bg-dark ">Type &nbsp; : &nbsp; {item.type}
                                    &nbsp;<span className="badge badge-info">{i + 1}</span>
                                </div>
                                <div className="card-body">
                                    <p>Description : &nbsp; {item.description} </p>
                                    <p>volume : &nbsp; {item.volume} </p>
                                </div>
                            </div>);
                        })}

                    </div>
                    <h4 className="shipment-header-text">Other info:</h4>
                    <table className="table table-dark table-striped">
                        <thead>
                            <tr>
                                <td>Mode</td>
                                <td>Type</td>
                                <td>Destination</td>
                                <td>Origin</td>
                            </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <td>{this.state.data.mode}</td>
                                <td>{this.state.data.type}</td>
                                <td>{this.state.data.destination}</td>
                                <td>{this.state.data.origin}</td>
                            </tr>
                        </tbody>
                    </table>

                    <h4 className="shipment-header-text">Services</h4>

                    <div className="card-group">

                        {this.state.data.cargo && this.state.data.services.map((item, i) => {
                            return (<div className="card bg-info" key={i}>

                                <div className="card-body">
                                    {item.type}
                                </div>
                            </div>);
                        })}

                    </div>

                </div >
            );
        } else if (this.state.error === 3) {
            return (
                <div className="error-message">
                    Either the rest url configured by is not correct or the record you are looking for is not found
                </div>
            );
        }
        return (<div>loading...</div>)

    }
}

export default ShipmentDetail;