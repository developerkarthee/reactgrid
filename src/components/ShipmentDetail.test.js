import React from 'react';
import {shallow} from 'enzyme';
import axios from 'axios';
import ShipmentDetail from './ShipmentDetail';
describe("test for shipment detail", () => {
    let mountedShipmentDetail;
    beforeEach(() => {
        jest.spyOn(axios, "get").mockImplementationOnce(()=>{
            return Promise.resolve(undefined);
        });
        mountedShipmentDetail = shallow(<ShipmentDetail match={{params :{id : 1}}}/>)
    });

    it('should render Shipment detail Component layout', () => {
        expect(mountedShipmentDetail.getElements()).toMatchSnapshot();
    });

    it('input text change event test', () => {
        
        const component =  mountedShipmentDetail;
        component.setState({ error: 2 })
        const mockEvent = {
            target: {
                value: 'prd1'
            }
        };
        expect(mountedShipmentDetail.find('#name').length).toEqual(1);
        mountedShipmentDetail.find('#name').simulate('change', mockEvent);
        expect(component.state("data").name).toEqual('prd1');
    });
});

