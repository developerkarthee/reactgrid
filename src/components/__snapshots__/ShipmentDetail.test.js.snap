// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for shipment detail should render Shipment detail Component layout 1`] = `
Array [
  <div
    className="error-message"
  >
    Either the rest url configured by is not correct or the record you are looking for is not found
  </div>,
]
`;
